tgintegration/containers/keyboard.py

```diff
@@ -1,91 +1,88 @@
 import itertools
 import re
 import weakref
-from typing import Union, List, Pattern, Optional, TYPE_CHECKING
+from typing import List, Optional, Pattern, TYPE_CHECKING, Union
 
-from pyrogram import Client
-from pyrogram.filters import chat, edited, text
-from pyrogram.raw.base.messages import BotCallbackAnswer
-from pyrogram.types import KeyboardButton, Message
-from pyrogram.types import InlineKeyboardButton
-
-from tgintegration import AwaitableAction
+from pyrogram import filters as f
+from pyrogram.raw.types.messages import BotCallbackAnswer
+from pyrogram.types import InlineKeyboardButton, KeyboardButton, Message
 
 if TYPE_CHECKING:
-    from tgintegration import Response
+    from tgintegration import BotController, Response
 
 
 class ReplyKeyboard:
     def __init__(
         self,
-        client: Client,
+        controller: "BotController",
         chat_id: Union[int, str],
         message_id: int,
         button_rows: List[List[KeyboardButton]],
     ):
-        self._client = weakref.proxy(client)
+        self._controller: BotController = weakref.proxy(controller)
```

Comment:
Why a weakref?

Reply:
My old me from two years ago didn't know what the heck he was doing. Good catch, removing!